### DISCUSSION TITLE
Excessive Check Count & Unrecognized Runs

### DISCUSSION THREAD

**Discussion by mohammed90** [community/community#26068] [Actions]
<p>Hello :wave:</p>
<p>We seem to have a problem. We have <a href="https://github.com/caddyserver/caddy/pull/3673" rel="nofollow noopener">PR updating a few Github Actions workflows</a>, and, somehow, this broke Github Actions recognition of runs and the badge is miscounting the checks too.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3f88ff7c537757c54a0efbdd414f3c15ed3a6680.png" data-download-href="/uploads/short-url/943B2Ic8Bn9bFWZ4bQLq80iSIyA.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2636183/181095057-e4aee48c-709b-4e50-9b2a-8d9e361af0b4.png" alt="image" data-base62-sha1="943B2Ic8Bn9bFWZ4bQLq80iSIyA" width="690" height="448" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3f88ff7c537757c54a0efbdd414f3c15ed3a6680_2_690x448.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/3f88ff7c537757c54a0efbdd414f3c15ed3a6680_2_1035x672.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/3f88ff7c537757c54a0efbdd414f3c15ed3a6680_2_1380x896.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3f88ff7c537757c54a0efbdd414f3c15ed3a6680_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1878×1220 283 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e19d903320b8ffcf0dd3a309f7a83521b86f98ed.png" data-download-href="/uploads/short-url/wbT0x8PyakR6GZOKa5PjCRGhb1b.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2636183/181095060-9c038632-97fe-486f-9de6-c1c1cbfe103d.png" alt="image" data-base62-sha1="wbT0x8PyakR6GZOKa5PjCRGhb1b" width="690" height="130" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e19d903320b8ffcf0dd3a309f7a83521b86f98ed_2_690x130.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e19d903320b8ffcf0dd3a309f7a83521b86f98ed_2_1035x195.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e19d903320b8ffcf0dd3a309f7a83521b86f98ed_2_1380x260.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e19d903320b8ffcf0dd3a309f7a83521b86f98ed_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1878×354 45.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>We aren’t entirely sure what’s causing this besides the hunch that maybe adding more Go versions somehow broke things. We tried turning it off and on again (close and re-open). We tried issuing a duplicate PR. Neither of those worked.</p>
<p>Any cues, help, or troubleshooting would be great :slightly_smiling_face:</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mohammed90">@mohammed90</a>,</p>
<p>This problem generally is due to the required checks do not run or are not triggered on the latest commit of the pull request.<br>
In your case, the three required checks ‘<strong>test (macos-latest, 1.14.x)</strong>’, ‘<strong>test (ubuntu-latest, 1.14.x)</strong>’ and ‘<strong>test (windows-latest, 1.14.x)</strong>’ should be the jobs in the workflow runs related to the pull request.</p>
<p>The workflow runs related to the PR generally include these:</p>
<ol>
<li>
<p>Workflow runs triggered by the events related to the pull request. The events include <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="nofollow noopener"><strong>pull_request</strong></a>, <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review" rel="nofollow noopener"><strong>pull_request_review</strong></a>, <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review_comment" rel="nofollow noopener"><strong>pull_request_review_comment</strong></a> and <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target" rel="nofollow noopener"><strong>pull_request_target</strong></a>.</p>
</li>
<li>
<p>Workflow runs triggered by <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push" rel="nofollow noopener"><strong>push</strong></a> event for the latest commit on the head branch (source branch) of the pull request.</p>
</li>
</ol>
<p>If the three required checks do not run or are not triggered on the latest commit of the pull request. You can navigate to the repository’s <strong>Settings</strong> &gt; <strong>Branches</strong> &gt; <strong>Branch protection rules</strong>. Open the and edit the rules you have set for the <strong>master</strong> branch. Disable the three required checks.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cf8bbee4090be5de5305d8fe5964b401ef3ea46f.png" data-download-href="/uploads/short-url/tC2eSV437Y3cU9VQbYgWaLcbVLx.png?dl=1" title="Branch-rules" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181095062-96d97cd2-c51d-4415-9f6f-d94658824b04.png" alt="Branch-rules" data-base62-sha1="tC2eSV437Y3cU9VQbYgWaLcbVLx" width="613" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cf8bbee4090be5de5305d8fe5964b401ef3ea46f_2_613x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/cf8bbee4090be5de5305d8fe5964b401ef3ea46f_2_919x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/cf8bbee4090be5de5305d8fe5964b401ef3ea46f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cf8bbee4090be5de5305d8fe5964b401ef3ea46f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Branch-rules</span><span class="informations">1014×827 51.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>After changing the rules, refresh or reopen the web page of the pull request, then you can see the three checks disappear from the list of the checks, and you can merge the PR.</p>
<p>To create, edit or delete the rules, you need the owner or the administrator permissions.</p>

**Comment by mohammed90** [comment]
<p>That did it. Thanks!</p>
